cmd/foxtrot/main.go

```diff
@@ -1,27 +1,61 @@
 package main
 
 import (
+	"bufio"
 	"log"
+	"net"
 	"net/http"
-	"os"
 
+	"foxygo.at/foxtrot/pkg/foxtrot"
+	"github.com/alecthomas/kong"
 	"github.com/gorilla/websocket"
+	_ "github.com/mattn/go-sqlite3"
 )
 
 func main() {
-	http.Handle("/", http.FileServer(http.Dir("static")))
-	http.HandleFunc("/ws", socketHandler)
+	cfg := &foxtrot.Config{}
+	kong.Parse(cfg, kong.Description("Foxtrot Server"))
 
-	port := os.Getenv("PORT")
-	if port == "" {
-		port = "8080"
+	mux := http.NewServeMux()
+	mux.Handle("/", http.FileServer(http.Dir("static")))
+	mux.HandleFunc("/ws", socketHandler)
```

Comment:
Nice. trying it out.

Reply:
Seen this? Might have also been an answer to previous problems? https://stackoverflow.com/a/43380025/661500

Reply:
so the following is an alternative. but I'm still most inclined to leave as is
less code, works with all standard mux shanangians. could be convinced otherwise though.
tell me again: the argument was if something wants to do comma,ok check on hijacker this will pass but on execution fail and that's ugly. yeah?
```go
	mux := http.NewServeMux()
	mux.Handle("/", http.FileServer(http.Dir("static")))
	if _, err := foxtrot.NewApp(cfg, mux); err != nil {
		log.Fatal(err)
	}
	h := logHTTP(mux)
	rootMux := http.NewServeMux()
	rootMux.HandleFunc("/ws", socketHandler)
	rootMux.Handle("/", h)

	port := ":8080"
	log.Printf("Listening on port %s", port)
	if err := http.ListenAndServe(port, rootMux); err != nil {
		log.Fatal(err)
	}
```

Reply:
actually, thought about it some more.... i think what you are suggesting is fine.

Reply:
Yeah, I think I've nested muxes before. I didn't think it was worth it here - to much guff and overhead for what would otherwise be a single if/else statement.

And about Hijacker, yeah, I think it's wrong to implement the interface only to have it fail. I don't think Go has a good solution here.

Reply:
ok, used yours, done.